remix-solidity/src/analysis/modules/staticAnalysisCommon.js

```diff
@@ -531,6 +558,33 @@ function isConstructor (node) {
   )
 }
 
+/**
+ * True if is block has top level binops (e.g. that are not assigned to anything, most of the time confused compare instead of assign)
+ * @node {ASTNode} some AstNode
+ * @return {bool}
+ */
+function isBlockWithTopLevelUnAssignedBinOp (node) {
+  return nodeType(node, exactMatch(nodeTypes.BLOCK)) && node.children && node.children.some(isBinaryOpInExpression) ||
+          isBlockLikeStatement(node) && node.children && node.children.some(isBinaryOpInExpression) // Second Case for if without braces
+}
+
+function isBlockLikeStatement (node) {
```

Comment:
Perhaps this could be a more generic `subScopeStatement`?

Reply:
what changes scoping wise with 0.5.0?

Reply:
It moves on to C++ scoping.

Reply:
But C++ still has blocks and statements that have either a block or a single statement underneath it (if, while, ...) I do not look at variables and if their in scope at all. I just check if there are bin ops that are never assigned to anything. If the variable assigned to is in scope is anyways checked by the compiler. So i thing we are fine unless i miss something.